buffer/src/main/java/io/netty/buffer/UnpooledUnsafeDirectByteBuf.java

```diff
@@ -95,7 +99,7 @@ protected UnpooledUnsafeDirectByteBuf(ByteBufAllocator alloc, ByteBuffer initial
         }
 
         this.alloc = alloc;
-        doNotFree = true;
+        doNotFree = !doNotFree;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Uninitialized read of doNotFree in new io.netty.buffer.UnpooledUnsafeDirectByteBuf(ByteBufAllocator, ByteBuffer, int, boolean) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=findbugs%3AUR_UNINIT_READ)